Sources/generate-symbol-graph/main.swift

```diff
@@ -637,6 +637,136 @@ let symbols: [SymbolGraph.Symbol] = supportedDirectives.map { directive in
+    var usage: String
+}
+
+let documentedCommands = [
+    DocumentedCommand(
+        name: "convert",
+        introduced: "5.5",
+        usage: """
+        docc convert \\
+          [<catalog-path>]
+          [--additional-symbol-graph-dir <symbol-graph-dir>] \\
+          [--output-dir <output-dir>]  \\
+          [<availability-options>] \\
+          [<diagnostic-options>] \\
+          [<source-repository-options>] \\
+          [<hosting-options>] \\
+          [<info-plist-fallbacks>] \\
+          [<feature-flags>] \\
+          [<other-options>]
+        """
+    ),
+    
+    DocumentedCommand(
+        name: "preview",
+        introduced: "5.5",
+        usage: """
+        docc preview \\
+          [--port <port-number>] \\
+          [<convert-options>]
+        """
+    ),
+    
+    DocumentedCommand(
+        name: "index",
+        introduced: "5.5",
+        usage: """
+        docc process-archive index \\
+          [<source-archive-path>] \\
+          --bundle-identifier <bundle-identifier> \\
+          [--verbose]
+        """
+    ),
+    
+    DocumentedCommand(
+        name: "transform-for-static-hosting",
+        introduced: "5.6",
+        usage: """
+        docc process-archive transform-for-static-hosting \\
+        [<source-archive-path>] \\
+        [--output-path <output-path>] \\
+        [--hosting-base-path <hosting-base-path>]
+        """
+    ),
+    
+    DocumentedCommand(
+        name: "emit-generated-curation",
+        introduced: "5.11",
+        usage: """
+        docc process-catalog emit-generated-curation \\
+          [<catalog-path>] \\
+          [--additional-symbol-graph-dir <symbol-graph-dir>] \\
+          [--output-path <output-path>] \\
+          [--from-symbol <symbol-link>] \\
+          [--depth <limit>]
+        """
+    ),   
+    
+    DocumentedCommand(
+        name: "init",
+        introduced: "5.11",
+        usage: """
+        docc init \\
+          --name <name> \\
+          --output-dir <output-dir> \\
+          --template <template-name>
+        """
+    ),
+]
+
+symbols += documentedCommands.map { command in
+    var scanner = CommandUsageScanner(command.usage[...])
+    let declarationFragments: [SymbolGraph.Symbol.DeclarationFragments.Fragment] = scanner.scanTokens().map {
+        switch $0.kind {
+        case .command:
+            return .init(kind: .identifier, spelling: $0.text, preciseIdentifier: nil)
+        case .text:
+            return .init(kind: .text, spelling: $0.text, preciseIdentifier: nil)
+        case .flag:
+            return .init(kind: .externalParameter, spelling: $0.text, preciseIdentifier: nil)
+        case .value:
+            return .init(kind: .typeIdentifier, spelling: $0.text, preciseIdentifier: nil)
+        }
+    }
+    
+    let subHeading = declarationFragments
+        .filter { $0.kind == .identifier }
+        .flatMap { [$0, .init(kind: .text, spelling: " ", preciseIdentifier: nil)] }
```

Comment:
Could you use `joined(separator:)` here instead? Seems like the perfect use case for it.